PWGCF/FEMTOSCOPY/FemtoDream/AliAnalysisTaskNanoPt.cxx

```diff
@@ -0,0 +1,368 @@
+
+const Float_t AliAnalysisTaskNanoPt::GetMass2sq(AliFemtoDreamTrack *track) {
+  Float_t p = track->GetP();
+  Float_t mass2sq = -999;
+  Float_t beta = track->GetbetaTOF();
+  if (!(beta > 0)) {
+    mass2sq = ((1 / (beta * beta)) - 1) * (p * p);
+  }
+  return mass2sq;
+}
+
+//-------------------------------------------------------UserCreateOutPut----------------------------------------------------------------------------------
+
+void AliAnalysisTaskNanoPt::UserCreateOutputObjects() {
+
+  fGTI = new AliVTrack *[fTrackBufferSize];
+
+  if (!fEvtCuts) {
+    AliError("No Event cuts \n");
+  } else {
+    fEvtCuts->InitQA();
+  }
+
+  if (!fProtonTrack) {
+    AliError("No Proton cuts \n");
+  } else {
+    fProtonTrack->Init();
+    fProtonRestMass = new TH2F("fProtonRestMass", "Proton", 36, 0.5, 4.05, 180, 0.2, 2.);
+    fProtonRestMass->GetXaxis()->SetTitle("pT(GeV)");
+    fProtonRestMass->GetYaxis()->SetTitle("m^2(Gev)^2");
+  }
+
+  if (!fAntiProtonTrack) {
+    AliError("No AntiProton cuts \n");
+  } else {
+    fAntiProtonTrack->Init();
+    fAntiProtonRestMass = new TH2F("fAntiProtonRestMass", "AntiProton", 36, 0.5, 4.05, 180, 0.2, 2.);
+    fAntiProtonRestMass->GetXaxis()->SetTitle("pT(GeV)");
+    fAntiProtonRestMass->GetYaxis()->SetTitle("m^2(Gev)^2");
+  }
+
+  if (!fDeuteronTrack) {
+    AliError("No Proton cuts \n");
+  } else {
+    fDeuteronTrack->Init();
+    fDeuteronRestMass = new TH2F("fDeuteronRestMass", "Deuteron", 36, 0.5, 4.05, 180, 0.2, 2.);
+    fDeuteronRestMass->GetXaxis()->SetTitle("pT(GeV)");
+    fDeuteronRestMass->GetYaxis()->SetTitle("m^2(Gev)^2");
+  }
+
+  if (!fAntiDeuteronTrack) {
+    AliError("No Proton cuts \n");
+  } else {
+    fAntiDeuteronTrack->Init();
+    fAntiDeuteronRestMass = new TH2F("fAntiDeuteronRestMass", "AntiDeuteron", 36, 0.5, 4.05, 180, 0.2, 2.);
+    fAntiDeuteronRestMass->GetXaxis()->SetTitle("pT(GeV)");
+    fAntiDeuteronRestMass->GetYaxis()->SetTitle("m^2(Gev)^2");
+  }
+
+  if (!fConfig) {
+    AliError("No Correlation Config \n");
+  } else {
+    fPartColl = new AliFemtoDreamPartCollection(fConfig,
+        fConfig->GetMinimalBookingME());
+    fPairCleaner = new AliFemtoDreamPairCleaner(2, 0,
+        fConfig->GetMinimalBookingME());
+  }
+
+  fEvent = new AliFemtoDreamEvent(false, true, GetCollisionCandidates(), false);
+
+  fTrack = new AliFemtoDreamTrack();
+  fTrack->SetUseMCInfo(fIsMC);
+
+  if (!fEvtCuts->GetMinimalBooking()) {
+    fEvtList = fEvtCuts->GetHistList();
+  } else {
+    fEvtList = new TList();
+    fEvtList->SetName("EventCuts");
+    fEvtList->SetOwner();
+  }
+
+
+  fProtonList = fProtonTrack->GetQAHists();
+  fProtonList->Add(fProtonRestMass);
+  fAntiProtonList = fAntiProtonTrack->GetQAHists();
+  fAntiProtonList->Add(fAntiProtonRestMass);
+  fDeuteronList = fDeuteronTrack->GetQAHists();
+  fDeuteronList->Add(fDeuteronRestMass);
+  fAntiDeuteronList = fAntiDeuteronTrack->GetQAHists();
+  fAntiDeuteronList->Add(fAntiDeuteronRestMass);
```

Comment:
maybe do this where you create the histograms, otherwise this might cause a crash